src/main/java/org/jboss/threads/ViewExecutor.java

```diff
@@ -361,7 +361,11 @@ public void run() {
                     }
                 }
             } finally {
-                thread = null;
+                // Must be synchronized with interrupt() to avoid acquiring the thread reference as work completes
+                // and interrupting the next task run by this thread which may not originate from this view.
+                synchronized (this) {
+                    thread = null;
```

Comment:
Note that in the case this TaskWrapper re-submits itself (line 354) it's possible execution has already begun on another thread, and this may null out that thread value.

Reply:
updated to resolve this case as well.